packages/opencensus-web-exporter-ocagent/src/api-types.ts

```diff
@@ -0,0 +1,531 @@
+   * resource set.
+   * If unset, the most recently set resource in the RPC stream applies. It is
+   * valid to never be set within a stream, e.g. when no resource info is known.
+   */
+  resource?: Resource;
+}
+
+/**
+ * Format for an HTTP/JSON response from a grpc-gateway for trace span exporter.
+ */
+export interface ExportTraceServiceResponse {}
+
+/** Resource information. */
+export interface Resource {
+  /** Type identifier for the resource. */
+  type?: string;
+
+  /** Set of labels that describe the resource. */
+  labels?: {[key: string]: string};
+}
+
+/**
+ * Identifier metadata of the Node (Application instrumented with OpenCensus)
+ * that connects to OpenCensus Agent.
+ * In the future we plan to extend the identifier proto definition to support
+ * additional information (e.g cloud id, etc.)
+ */
+export interface Node {
+  /** Identifier that uniquely identifies a process within a VM/container. */
+  identifier?: ProcessIdentifier;
+
+  /** Information on the OpenCensus Library that initiates the stream. */
+  libraryInfo?: LibraryInfo;
+
+  /** Additional information on service. */
+  serviceInfo?: ServiceInfo;
+
+  /** Additional attributes. */
+  attributes?: {[key: string]: string};
+}
+
+export interface LibraryInfo {
+  /**
+   * TODO(draffensperger): Get additional language option for web JavaScript,
+   * and then set the library version here.
+   */
+
+  exporterVersion?: string;
+
+  coreLibraryVersion?: string;
+}
+
+/**
+ * Identifier that uniquely identifies a process within a VM/container.
+ * For OpenCensus Web, this identifies the domain name of the site.
+ */
+export interface ProcessIdentifier {
+  /**
+   * The host name. Usually refers to the machine/container name.
+   * For example: os.Hostname() in Go, socket.gethostname() in Python.
+   * This will be the value of `window.location.host` for OpenCensus Web.
+   */
+  hostName?: string;
+
+  /** Process id. Not used in OpenCensus Web. */
+  pid?: number;
+
+  /** Start time of this ProcessIdentifier. Represented in epoch time.. */
+  startTimestamp?: string;
+}
+
+/** Additional service information. */
+export interface ServiceInfo {
+  /** Name of the service. */
+  name?: string;
+}
+
+/** Type for the `tracestate`  */
+export type TraceState = {
+  [key: string]: string
+};
+
+/**
+ * A span represents a single operation within a trace. Spans can be nested to
+ * form a trace tree. Often, a trace contains a root span that describes the
+ * end-to-end latency, and one or more subspans for its sub-operations. A trace
+ * can also contain multiple root spans, or none at all. Spans do not need to be
+ * contiguous - there may be gaps or overlaps between spans in a trace.  The
+ * next id is 16. TODO(bdrutu): Add an example.
+ */
+export interface Span {
+  /**
+   * A unique identifier for a trace. All spans from the same trace share the
+   * same `trace_id`. The ID is a 16-byte array.  This field is required.
+   */
+  traceId?: string;
```

Comment:
Since the field is required, should this property (and the `spanId` below) be required?